In-place upgrade in an airgapped server

parthaa opened the pull request:
This  commit adds the steps that would enable customer to upgrade from
6.10.0 -> 6.10.z in an airgapped downstream server.


Cherry-pick into:

* [x] Foreman 3.1
* For Foreman 3.0 or older, file a separate PR request

<!---
Thank you for contributing to Foreman documentation. Make sure to read README
for the documentation standards. Set cherry-pick github label to mark this
contribution for cherry picking and check which version do you need with [x].
-->

melcorr commented:
Thank you for this @parthaa 
i have this on my TODO list for Monday.

parthaa commented:
@melcorr have a feeling I need to add the disconnected upgrade  at a different spot rather than here. Will message you after I redo

melcorr commented:
@parthaa I had a bit of a crazy week, but I am here now and reviewing.

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -70,120 +60,209 @@ If you lose connection to the command shell where the upgrade command is running
 +
-. In the configuration file, replace `My_Organization` in the `baseurl` with the correct organization label.
-To obtain the organization label, enter the command:
+* In the configuration file, replace 
+** `/etc/pki/katello/certs/org-debug-cert.pem` in `sslclientcert` and `sslclientkey` with the location of the downloaded organization debug certificate.
+** `satellite.example.com` with correct FQDN 
+** `My_Organization` with the correct organization label in the `baseurl`. To obtain the organization label, enter the command:
```
```suggestion
. In the configuration file, replace `/etc/pki/katello/certs/org-debug-cert.pem` in `sslclientcert` and `sslclientkey` with the location of the downloaded organization debug certificate.
. Update `{foreman-example-com}` with correct FQDN for your deployment. 
. Replace `My_Organization` with the correct organization label in the `baseurl`. 
To obtain the organization label, enter the following command:
```

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -70,120 +60,209 @@ If you lose connection to the command shell where the upgrade command is running
+# hammer organization list
+----
+
+* Run the `reposync` command as follows:
```
```suggestion
. Enter the `reposync` command:
```

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -70,120 +60,209 @@ If you lose connection to the command shell where the upgrade command is running
+ -r {RepoRHEL7ServerSoftwareCollections}
 ----
+
+* This will download the contents of these repositories from the Upstream Satellite and store it in the directory `~/satellite-repos`.
```
```suggestion
This downloads the contents of the repositories from the connected {ProjectServer} and stores them in the directory `~/{Project}-repos`.
```

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -70,120 +60,209 @@ If you lose connection to the command shell where the upgrade command is running
 ----
+
+* This will download the contents of these repositories from the Upstream Satellite and store it in the directory `~/satellite-repos`.
+* `reposync` in RHEL 7 at this point does not download the yum metadata. It downloads just the rpms. Due to we would need to manually run `createrepo .` in each sub-directory of `satellite-repos`. You may use the following to automate this. 
```
```suggestion
The `reposync` command in {RHEL} 7 does not download the RPMs but not the Yum metadata. 
Because of this, you must manually run `createrepo .` in each sub-directory of `{Project}-repos`. 
You can use the following command to automate this: 
```

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -70,120 +60,209 @@ If you lose connection to the command shell where the upgrade command is running
 +
+[options="nowrap" subs="attributes"]
+----
+# cd ~/satellite-repos
```
```suggestion
# cd ~/{Project}-repos
```

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -70,120 +60,209 @@ If you lose connection to the command shell where the upgrade command is running
+----
 
-. Ensure that the `rubygem-foreman_maintain` package that provides `{foreman-maintain}` is installed and up to date:
+* Verify that the rpm's have been downloaded and the repodata directory is generated in each of the subdirectories of `~/satellite-repos`.
```
```suggestion
Verify that the RPMs have been downloaded and the repository data directory is generated in each of the sub-directories of `~/{Project}-repos`.
```

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -70,120 +60,209 @@ If you lose connection to the command shell where the upgrade command is running
 
-. Ensure that the `rubygem-foreman_maintain` package that provides `{foreman-maintain}` is installed and up to date:
+* Verify that the rpm's have been downloaded and the repodata directory is generated in each of the subdirectories of `~/satellite-repos`.
+* Archive the contents of the directory 
```
```suggestion
. Archive the contents of the directory. 
```

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -70,120 +60,209 @@ If you lose connection to the command shell where the upgrade command is running
 ----
 
-. Check the available versions to confirm the next minor version is listed:
+* We can now use the generated `satellite-repos.tgz` file to upgrade in the Downstream {ProjectServer}.
```
```suggestion
. Use the generated `{Project}-repos.tgz` file to upgrade in the disconnected {ProjectServer}.
```

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -70,120 +60,209 @@ If you lose connection to the command shell where the upgrade command is running
-. Check the available versions to confirm the next minor version is listed:
+* We can now use the generated `satellite-repos.tgz` file to upgrade in the Downstream {ProjectServer}.
+
+*On the Downstream {ProjectServer}*
```
```suggestion
Perform the following steps on the disconnected {ProjectServer}.
```

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -70,120 +60,209 @@ If you lose connection to the command shell where the upgrade command is running
+
+*On the Downstream {ProjectServer}*
+
+* Copy the generated `satellite-repos.tgz` file from above to Downstream {ProjectServer}.
```
```suggestion
. Copy the generated `{Project}-repos.tgz` file to your disconnected {ProjectServer}.
```

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -70,120 +60,209 @@ If you lose connection to the command shell where the upgrade command is running
+*On the Downstream {ProjectServer}*
+
+* Copy the generated `satellite-repos.tgz` file from above to Downstream {ProjectServer}.
+* Extract the archive to anywhere accessible by root. In the example below we are extracting it at `/root`
```
```suggestion
. Extract the archive to anywhere accessible by the `root` user. 
In the following example `/root` is the extraction location. 
```

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -70,120 +60,209 @@ If you lose connection to the command shell where the upgrade command is running
+# cd /root
+# tar zxf satellite-repos.tgz
+----
+* Create a new yum configuration file as follows:
```
```suggestion
+
. Create a Yum configuration file:
```

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -70,120 +60,209 @@ If you lose connection to the command shell where the upgrade command is running
++
+[options="nowrap" subs="attributes"]
+----
+# vi /etc/yum.repos.d/redhat-local.repo
```
```suggestion
# vi /etc/yum.repos.d/redhat-local.repo
----
+
```

ekohl replied:
Alternative: don't suggest an editor to the user?

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -70,120 +60,209 @@ If you lose connection to the command shell where the upgrade command is running
+[options="nowrap" subs="attributes"]
+----
+# vi /etc/yum.repos.d/redhat-local.repo
+
```
```suggestion
. Add the following repository information to the file:
+
[options="nowrap" subs="attributes"]
----
```

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -70,120 +60,209 @@ If you lose connection to the command shell where the upgrade command is running
+[{RepoRHEL7ServerAnsible}]
+name=Ansible {SatelliteAnsibleVersion} RPMs for Red Hat Enterprise Linux 7 Server x86_64
+baseurl=file:///root/satellite_repos/rhel-7-server-ansible-2.9-rpms
+>>>>>>> 072ea2d6 (In-place upgrade in an airgapped server)
```
This looks like garbage from a conflict.

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -70,120 +60,209 @@ If you lose connection to the command shell where the upgrade command is running
+
+[{RepoRHEL7Server}]
+name=Red Hat Enterprise Linux 7 Server RPMs x86_64
+baseurl=file:///root/satellite_repos/rhel-7-server-rpms
```
```suggestion
baseurl=file:///root/{Project}_repos/rhel-7-server-rpms
```
Can you confirm the file name for Foreman and Satellite?

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -70,120 +60,209 @@ If you lose connection to the command shell where the upgrade command is running
+
+[{RepoRHEL7ServerSatelliteServerProductVersion}]
+name={ProjectNameX} for RHEL 7 Server RPMs x86_64
+baseurl=file:///root/satellite_repos/rhel-7-server-satellite-6.10-rpms
```
Please use the attribute here also.

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -70,120 +60,209 @@ If you lose connection to the command shell where the upgrade command is running
+baseurl=file:///root/satellite_repos/rhel-7-server-satellite-6.10-rpms
+enabled=1
+
+[{RepoRHEL7ServerSatelliteMaintenanceProductVersion}]
+name={ProjectName} Maintenance 6 for RHEL 7 Server RPMs x86_64
+baseurl=file:///root/satellite_repos/rhel-7-server-satellite-maintenance-6-rpms
+enabled=1
+
+[{RepoRHEL7ServerSoftwareCollections}]
+name=Red Hat Software Collections RPMs for Red Hat Enterprise Linux 7 Server x86_64
+baseurl=file:///root/satellite_repos/rhel-server-rhscl-7-rpms
+enabled=1
+----
```
Please use the attributes everywhere. Please please please.

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -70,120 +60,209 @@ If you lose connection to the command shell where the upgrade command is running
+enabled=1
+----
++
+* In the configuration file, replace the `/root/satellite_repos` with the extracted location (if that is in a different directory.)
```
```suggestion
. In the configuration file, replace the `/root/{Project}_repos` with the extracted location.
```

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -70,120 +60,209 @@ If you lose connection to the command shell where the upgrade command is running
 Review the results and address any highlighted error conditions before performing the upgrade.
 
-. Because of the lengthy update time, use a utility such as `screen` to suspend and reattach a communication session.
+* Because of the lengthy update time, use a utility such as `screen` to suspend and reattach a communication session.
```
```suggestion
Because of the lengthy update time, use a utility such as `screen` to suspend and reattach a communication session.
```

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -70,120 +60,209 @@ If you lose connection to the command shell where the upgrade command is running
 If you lose connection to the command shell where the upgrade command is running, you can see the logged messages in the `{installer-log-file}` file to check if the process completed successfully.
 
-. Perform the upgrade:
+* Perform the upgrade:
```
These completely unnecessary changes are truly painful to review. 
It was not needed at all. 
```suggestion
. Perform the upgrade:
```

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -70,120 +60,209 @@ If you lose connection to the command shell where the upgrade command is running
 ----
 
-. Check when the kernel packages were last updated:
+* Check when the kernel packages were last updated:
```
```suggestion
. Check when the kernel packages were last updated:
```

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -70,120 +60,209 @@ If you lose connection to the command shell where the upgrade command is running
 ----
 
-. Optional: If a kernel update occurred since the last reboot, stop the `{foreman-maintain}` services and reboot the system:
+* Optional: If a kernel update occurred since the last reboot, stop the `{foreman-maintain}` services and reboot the system:
```
```suggestion
. Optional: If a kernel update occurred since the last reboot, stop the `{foreman-maintain}` services and reboot the system:
```

melcorr reviewed (changes requested):
@parthaa 
Thanks for making the changes. 
Please, please, please, don't make random and unnecessary changes to file formatting. 
We write in sequential steps when a user performs steps one after the other. The additions of bullets everywhere makes no sense. 
Please, please, please, preview locally before pushing again. Please think of Foreman and Satellite scenarios. There's a removal of an if statement to exclude this from the upstream docset because at the time of the Katello 4.0 release, there were many items deemed to be Satellite only. 
Please, please, please implement every piece of feedback I have given. 
And thank you again for writing these docs.

parthaa commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -1,17 +1,7 @@
-If you customize configuration files, manually or use a tool such as Hiera, these customizations are overwritten when the installation script runs during upgrading or updating.
-You can use the `--noop` option with the {foreman-installer} script to test for changes.
-For more information, see the Red Hat Knowledgebase solution https://access.redhat.com/solutions/3351311[How to use the noop option to check for changes in {Project} config files during an upgrade].
+= Updating {ProjectServer} =
```
Hmm look here -> https://partha.fedorapeople.org/docs/build/Upgrading_and_Updating/index-katello.html#updating_satellite_server_to_next_minor_version looks valid to me there ?  

It may be incorrect according to ascii doc guide. Going to check and update.

parthaa replied:
I think you are right going to apply

melcorr replied:
Cool. Glad it works.  We don't do that in general, just use the = 

THANK YOU SO MUCH for previewing.

parthaa commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -70,120 +60,209 @@ If you lose connection to the command shell where the upgrade command is running
 # reboot
 ----
 
-ifdef::satellite[]
```
I see what you are saying. The main point of this pr is to rewrite the disconnected in-place foreman/satellite upgrade. I'll just add the `ifdef::satellite[]` and hide this from the user.  Basically that section just did sat 6.9 and instructions are totally different for 6.10.

melcorr replied:
Are these relevant for Satellite 7? If not, this should be pointed towards the 2.5 branch ?

parthaa replied:
yeh relevant for sat 6.10 and above

melcorr replied:
Perfect, thanks!

parthaa commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -70,120 +60,209 @@ If you lose connection to the command shell where the upgrade command is running
+* Create a new configuration file as follows:
 +
-[options="nowrap" subs="attributes"]
+[options="nowrap" subs="attributes"]guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc
```
Interesting the preview didn't show this.  Will address

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -70,120 +60,209 @@ If you lose connection to the command shell where the upgrade command is running
+*On the Upstream {ProjectServer}*
+
+* Ensure that you have synchronized the following repositories in your Upstream {ProjectServer}.
+** {RepoRHEL7ServerAnsible}
+** {RepoRHEL7Server}
+** {RepoRHEL7ServerSatelliteServerProductVersion}
+** {RepoRHEL7ServerSatelliteMaintenanceProductVersion}
+** {RepoRHEL7ServerSoftwareCollections}  
```
We have editing standards that I am trying to implement in this PR. 
We don't use these sub bullets in these docs. 
Your entry was fine but the formatting was not. 
I think it is best to use the variables. 
If you want use a single bulleted list.

parthaa commented:
Thanks for the review. I think I ve addressed all your comments and suggestions. Please let me know if I missed anything. 

Once you are happy with the formatting I am going to ask a katello dev/ qe to verify the steps

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -71,30 +71,26 @@ If you lose connection to the command shell where the upgrade command is running
-For more information about configuring download policies, see {ContentManagementDocURL}changing_the_download_policy_for_a_repository[Changing a download policy for a repository] in the _Content Management guide_.
-+
-* Ensure no Red Hat repositories are enabled by entering the command:
+This section describes the steps needed to update in an Air-gapped Disconnected setup where the Upstream {ProjectServer} (which sychronizes content from CDN) is air gapped from a disconnected {ProjectServer}.
```
Please use connected/disconnected and not upstream/downstream. Please!

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -103,54 +99,156 @@ For more information about configuring download policies, see {ContentManagement
 +
+[options="nowrap" subs="attributes"]
+----
+# cd ~
```
We separate each command into separate steps. 
So here we'd need a command, "Return to your previous location" or something like that and the cd - alone. 
cd -d doesn't generally need root.

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -103,54 +99,156 @@ For more information about configuring download policies, see {ContentManagement
+[options="nowrap" subs="attributes"]
+----
+# cd ~
+# reposync --delete --download-metadata -p {Project}-repos -n \
```
Separate procedure to introduce this.

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -159,7 +257,7 @@ On the first use of this command, `{foreman-maintain}` prompts you to enter the
 +
```
```suggestion

```
Here too. 
Preview locally and make sure there are no + signs.

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -159,7 +257,7 @@ On the first use of this command, `{foreman-maintain}` prompts you to enter the
+Because of the lengthy update time, use a utility such as `screen` to suspend and reattach a communication session.
 You can then check the upgrade progress without staying connected to the command shell continuously.
 For more information about using the screen command, see link:https://access.redhat.com/articles/5247[How do I use the screen command?] article in the _Red{nbsp}Hat Knowledge{nbsp}Base_.
 +
```
```suggestion

```
This needs to be removed as it is breaking the syntax.

melcorr reviewed (changes requested):
Hey @parthaa 
This is a big improvement. 
I read through there and there are still some outstanding issues. 
The order of steps is disrupted by some broken ascii threading. A misplaced + etc. It's an easy fix. 
The heading need to change if you need that subtitle. Either delete the subtitle or make it the real title. 
Once these are addressed, you can pass it over to the devs :)

ekohl commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -103,54 +99,156 @@ For more information about configuring download policies, see {ContentManagement
+[options="nowrap" subs="attributes"]
+----
+# cd ~/{Project}-repos
+# for directory in $(ls -d */)
```
Using `ls` in loops this way is considered a bad practice. The user may have other weird funky aliases. Doesn't this also work?
```suggestion
# for directory in */
```

ekohl commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -71,30 +71,26 @@ If you lose connection to the command shell where the upgrade command is running
 
 ifdef::satellite[]
-= Updating Disconnected {ProjectServer}
+= Updating Disconnected {ProjectServer} 
```
Trailing whitespace here
```suggestion
= Updating Disconnected {ProjectServer}
```

parthaa commented:
BTW how do I get the `satellite`  version. I tried `make html BUILD=satellite` and it still wont show the section under `ifdef::satellite[]`. The only way I have been able to preview is just commenting the `ifdef::satellite` before test and run `make html BUILD=katello`

ekohl commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -7,7 +7,7 @@
 * Ensure that you have synchronized {ProjectServer} repositories for {Project}, {SmartProxy}, and {project-client-name}.
 * Ensure each external {SmartProxy} and Content Host can be updated by promoting the updated repositories to all relevant Content Views.
-
+~/{Project}-repos
```
This looks like an odd thing. Copy paste error?

parthaa replied:
yeh correcting

parthaa commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -7,7 +7,7 @@
 * Ensure that you have synchronized {ProjectServer} repositories for {Project}, {SmartProxy}, and {project-client-name}.
 * Ensure each external {SmartProxy} and Content Host can be updated by promoting the updated repositories to all relevant Content Views.
-
+~/{Project}-repos
```
```suggestion
```

parthaa commented:
> I can't reproduce issues with the specific build. For me this works: `make BUILD=satellite html browser`.

that worked. Thanks!. I think I had `make html BUILD=satellite` which gave me a different result.

parthaa commented:
checkout  -> https://partha.fedorapeople.org/docs/build/Upgrading_and_Updating/index-satellite.html#upgrading_a_disconnected_satellite

melcorr commented:
Hey @parthaa 
Thank you so much for your patience. 
One remaining item from my last review. 
This subtitle cannot remain: https://drive.google.com/file/d/1MQ5UOtKLqZUS77DQpkfvrokS5TS3SwRT/view?usp=sharing
Either edit the title above it, or delete it.
And then we are done.

ekohl commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -7,7 +7,6 @@
 
 * Ensure that you have synchronized {ProjectServer} repositories for {Project}, {SmartProxy}, and {project-client-name}.
 * Ensure each external {SmartProxy} and Content Host can be updated by promoting the updated repositories to all relevant Content Views.
-
```
Probably better to keep this line in.

ekohl commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -103,54 +98,155 @@ For more information about configuring download policies, see {ContentManagement
+----
+[{RepoRHEL7ServerAnsible}]
+name=Ansible {SatelliteAnsibleVersion} RPMs for Red Hat Enterprise Linux 7 Server x86_64
+baseurl=file:///root/{Project}-repos/rhel-7-server-ansible-2.9-rpms
```
```suggestion
baseurl=file:///root/{Project}-repos/{RepoRHEL7ServerAnsible}
```

ekohl commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -103,54 +98,155 @@ For more information about configuring download policies, see {ContentManagement
+
+[{RepoRHEL7Server}]
+name=Red Hat Enterprise Linux 7 Server RPMs x86_64
+baseurl=file:///root/{Project}-repos/rhel-7-server-rpms
```
```suggestion
baseurl=file:///root/{Project}-repos/{RepoRHEL7Server}
```

ekohl commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -103,54 +98,155 @@ For more information about configuring download policies, see {ContentManagement
+
+[{RepoRHEL7ServerSatelliteServerProductVersion}]
+name={ProjectNameX} for RHEL 7 Server RPMs x86_64
+baseurl=file:///root/{Project}-repos/rhel-7-server-satellite-6.10-rpms
```
```suggestion
baseurl=file:///root/{Project}-repos/{RepoRHEL7ServerSatelliteServerProductVersion}
```

ekohl commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -103,54 +98,155 @@ For more information about configuring download policies, see {ContentManagement
+
+[{RepoRHEL7ServerSatelliteMaintenanceProductVersion}]
+name={ProjectName} Maintenance 6 for RHEL 7 Server RPMs x86_64
+baseurl=file:///root/{Project}-repos/rhel-7-server-satellite-maintenance-6-rpms
```
```suggestion
baseurl=file:///root/{Project}-repos/{RepoRHEL7ServerSatelliteMaintenanceProductVersion}
```

ekohl commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -103,54 +98,155 @@ For more information about configuring download policies, see {ContentManagement
 
+[{RepoRHEL7ServerSoftwareCollections}]
+name=Red Hat Software Collections RPMs for Red Hat Enterprise Linux 7 Server x86_64
+baseurl=file:///root/{Project}-repos/rhel-server-rhscl-7-rpms
```
```suggestion
baseurl=file:///root/{Project}-repos/{RepoRHEL7ServerSoftwareCollections}
```

ekohl commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -103,54 +98,155 @@ For more information about configuring download policies, see {ContentManagement
 [{RepoRHEL7ServerAnsible}]
 name=Ansible {SatelliteAnsibleVersion} RPMs for Red Hat Enterprise Linux 7 Server x86_64
-baseurl=file:///var/lib/pulp/published/yum/https/repos/My_Organization/Library/content/dist/rhel/server/7/7Server/x86_64/ansible/2.9/os/
+baseurl=https://{foreman-example-com}/pulp/content/My_Organization/Library/content/dist/rhel/server/7/7Server/x86_64/ansible/2.9/os/
```
Why hardcode x86_64 and not use `$basearch`? Also, instead of `7Server` I believe you can write `$releasever` in yum repo files. Sadly I don't think there's a macro in RHEL that resolves to `7` so you can't make the path fully dynamic.
```suggestion
baseurl=https://{foreman-example-com}/pulp/content/My_Organization/Library/content/dist/rhel/server/7/$releasever/$basearch/ansible/{SatelliteAnsibleVersion}/os/
```

This should get you closer to preparing for supporting RHEL 8.

ekohl commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -103,54 +98,155 @@ For more information about configuring download policies, see {ContentManagement
 [{RepoRHEL7ServerSatelliteServerProductVersion}]
 name={ProjectNameX} for RHEL 7 Server RPMs x86_64
-baseurl=file:///var/lib/pulp/published/yum/https/repos/My_Organization/Library/content/dist/rhel/server/7/7Server/x86_64/satellite/{ProjectVersion}/os/
+baseurl=https://{foreman-example-com}/pulp/content/My_Organization/Library/content/dist/rhel/server/7/7Server/x86_64/satellite/6.10/os/
```
Please keep using `{ProjectVersion}` and don't hardcode versions.

ekohl replied:
This is still here

parthaa commented:
@ekohl++ and @melcorr++  - excellent suggestions :). Thank you

rmynar commented:
I tried to update Satellite 6.10.0 to 6.10.2 using these steps, which went successfully, but following changes are needed:
- The _createrepo_ is not usually installed, so it should be mentioned how to install manually.
- satellite-maintain fails on _[check-upstream-repository]_ and _[repositories-validate]_, because we're using local repositories on disconnected satellite, so these checks need to be whitelisted `--whitelist="check-upstream-repository,repositories-validate"`

parthaa commented:
@rmynar++

@melcorr @ekohl  Updated the steps in my last commit based on @rmynar  's suggestions. Can you guys take a quick look?

ekohl commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -73,103 +73,195 @@ If you lose connection to the command shell where the upgrade command is running
+sslcacert = /etc/pki/katello/certs/katello-server-ca.crt
 
 [{RepoRHEL7ServerSatelliteMaintenanceProductVersion}]
 name={ProjectName} Maintenance 6 for RHEL 7 Server RPMs x86_64
```
This is OK for a cherry pick to an older version but heads up that this will change for Satellite 7 and technically master should already be following 7.

ekohl commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -73,103 +73,195 @@ If you lose connection to the command shell where the upgrade command is running
++
+[options="nowrap" subs="attributes"]
+----
+# {foreman-maintain} packages install createrepo
```
```suggestion
# {package-install-project} createrepo
```

ekohl reviewed (changes requested):
I only looked at use of various macros.

ekohl reviewed (approved):
From a macro point of view I have no objection to merging this, with the note that this is only appropriate for a cherry pick. Soon it'll need an update to also support RHEL 8.

parthaa commented:
@melcorr can you take a look :)

melcorr commented:
@parthaa sure.   On it now.

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -73,103 +73,195 @@ If you lose connection to the command shell where the upgrade command is running
+Because of the lengthy update time, use a utility such as `screen` to suspend and reattach a communication session.
 You can then check the upgrade progress without staying connected to the command shell continuously.
 For more information about using the screen command, see link:https://access.redhat.com/articles/5247[How do I use the screen command?] article in the _Red{nbsp}Hat Knowledge{nbsp}Base_.
-+
```
Losing these + signs starts a new procedure. 
Can you play around with one here and above.

melcorr commented on guides/doc-Upgrading_and_Updating/topics/updating_satellite_server_to_next_minor_version.adoc:
```diff
@@ -73,103 +73,195 @@ If you lose connection to the command shell where the upgrade command is running
 ----
 +
 Review the results and address any highlighted error conditions before performing the upgrade.
 
```
```suggestion
+
```
And below. 
Can you preview locally and ensure this all flows as one procedure?

parthaa replied:
Addressed this comment by making it a `.` instead of `+` to be consistent with what was  on line 45.
Check out 
https://partha.fedorapeople.org/docs/build/Upgrading_and_Updating/index-satellite.html#_updating_disconnected_satelliteserver 
and let me know.

melcorr reviewed (changes requested):
@parthaa one tiny formatting issue and we're good. 
Can you please confirm what versions of Katello/Satellite this applies to?

parthaa commented:
> Can you please confirm what versions of Katello/Satellite this applies to?

Foreman 2.5/Sat 6.10 and above.

melcorr commented:
@parthaa I am happy to merge this. Can you create a PR against 2.5 because things have diverged so much that it is not possible to cherry pick.

parthaa commented:
> @parthaa I am happy to merge this. Can you create a PR against 2.5 because things have diverged so much that it is not possible to cherry pick.

Created here  https://github.com/theforeman/foreman-documentation/pull/1037
